Разбивка компонента «Конструктор пиццы» на мелкие компоненты

iext opened the pull request:
---
:mortar_board: [Разбивка компонента «Конструктор пиццы» на мелкие компоненты](https://up.htmlacademy.ru/vue/1/user/1235117/tasks/3)

iext committed:
Убирает лишний $option из объекта

iext committed:
Убирает изменения в pizza.json и нормализует данные

iext committed:
Создаёт AppLayout с выносом хедера в него

iext committed:
Удаляет лишние index.js файлы

iext committed:
Выполняет нормализацию данных

iext committed:
Выделяет общие компоненты

iext committed:
Создаёт компоненты модуля Builder

Djaler commented on src/frontend/src/common/components/ItemCounter.vue:
```diff
@@ -0,0 +1,70 @@
+  },
+  data() {
+    return {
+      counterValue: JSON.parse(this.value),
```
тебе тут не нужно локальное состояние. используй везде value, приходящий в проп

Djaler commented on src/frontend/src/common/components/ItemCounter.vue:
```diff
@@ -0,0 +1,70 @@
+  methods: {
+    counterPlus() {
+      this.counterValue += 1;
+      this.$emit("plus", this.counterValue);
```
только вот тут, конечно, не нужно само значение менять, нужно просто кидать событие с `value + 1`

Djaler commented on src/frontend/src/common/components/ItemCounter.vue:
```diff
@@ -0,0 +1,70 @@
+    },
+    counterMinus() {
+      this.counterValue -= 1;
+      this.$emit("minus", this.counterValue);
```
ну и тебе тут не нужно 2 разных события, раз ты всё равно передаёшь новое значение. достаточно одного

Djaler commented on src/frontend/src/modules/builder/components/BuilderDoughSelector.vue:
```diff
@@ -0,0 +1,46 @@
+      <radio-button
+        v-for="dough in doughs"
+        :key="dough.value"
+        :name="'dough'"
```
в таких ситуациях можно делать просто `name="dough"`

Djaler commented on src/frontend/src/modules/builder/components/BuilderDoughSelector.vue:
```diff
@@ -0,0 +1,46 @@
+  },
+  data() {
+    return {
+      selectedDough: "light",
```
данные о выбранном тесте хорошо бы прокидывать сюда пропсом, чтоб был один источник истины

Djaler commented on src/frontend/src/modules/builder/components/BuilderIngredientsSelector.vue:
```diff
@@ -0,0 +1,96 @@
+            :key="ingredient.value"
+            class="ingridients__item"
+          >
+            <app-drag :transferData="ingredient" :draggable="true">
```
draggable нужно отключать при достижении максимального количества ингредиента

Djaler commented on src/frontend/src/modules/builder/components/BuilderIngredientsSelector.vue:
```diff
@@ -0,0 +1,96 @@
+    },
+
+    addIngredient(event, ingredient) {
+      ingredient.count = event;
```
вот тут ты неявно меняешь значение ingredient, которое пришло из пропса ingredients. Так делать нельзя, нужно данные менять только там, где они хранятся. Кидай в событии всё что нужно, чтобы в родительском компоненте обновить данные

Djaler commented on src/frontend/src/modules/builder/components/BuilderPizzaView.vue:
```diff
@@ -0,0 +1,87 @@
+        } else if (countIngredient === 3) {
+          classes.push("pizza__filling--third");
+        }
+        return classes.join(" ");
```
можно возвращать и массив, вью позволяет так делать

Djaler commented on src/frontend/src/modules/builder/components/BuilderPizzaView.vue:
```diff
@@ -0,0 +1,87 @@
+      let ingredients = [];
+      return this.ingredients.map((ingredient) => {
+        let classes = [`pizza__filling--${ingredient.value}`];
+        ingredients.push(ingredient.value);
```
а зачем тебе тут ещё снаружи map какой-то массив ингредиентов? map не должен делать никаких сайд-эффектов

iext committed:
Исправляет замечания

- Добавлены локальные стили для компонентов
- draggable отключается при достижении максимального кол-ва элементов
- Исправлена стилизация пиццы
- Заменены локальные состояния на value, приходящие в props
- Исправлено неявное изменение количества ингредиентов

Djaler commented on src/frontend/src/modules/builder/components/BuilderPizzaView.vue:
```diff
@@ -0,0 +1,86 @@
+  },
+  computed: {
+    ingredientsClasses() {
+      return (number) => {
```
это уже тогда стоит сделать методом

Djaler commented on src/frontend/src/views/Index.vue:
```diff
@@ -179,32 +55,99 @@
+      let index = this.ingredients.findIndex(
+        (item) => item.value === event.value
+      );
+      this.ingredients[index].count = event.count;
```
можно вместо findIndex делать find, чтоб проще было

Djaler commented on src/frontend/src/views/Index.vue:
```diff
@@ -179,32 +55,99 @@
+        (item) => item.value === event.value
+      );
+      this.ingredients[index].count = event.count;
+      this.ingredients.splice(index, 1, this.ingredients[index]);
```
ну и последняя строка тут не нужна, так как ты уже изменил данные в this.ingredients
